server/src/storage/time_series.rs

```diff
@@ -0,0 +1,212 @@
+/* Copyright 2022 The MiniModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Support for managing multiple compressed segments from the same time series.
+
+use std::fs;
+use std::io::ErrorKind::Other;
+use std::sync::Arc;
+
+use datafusion::arrow::datatypes::{ArrowPrimitiveType, DataType, Field, Schema};
+use datafusion::arrow::record_batch::RecordBatch;
+
+use crate::storage::write_batch_to_parquet;
+use crate::types::{ArrowTimestamp, ArrowValue, TimestampArray};
+
+/// A single compressed time series, containing one or more compressed segments and providing
+/// functionality for appending segments and saving all segments to a single Parquet file.
+pub struct CompressedTimeSeries {
+    /// Compressed segments that make up the sequential compressed data of the time series.
+    compressed_segments: Vec<RecordBatch>,
+    /// Continuously updated total sum of the size of the compressed segments.
+    pub size_in_bytes: usize,
+}
+
+impl CompressedTimeSeries {
+    pub fn new() -> Self {
+        Self {
+            compressed_segments: Vec::new(),
+            size_in_bytes: 0,
+        }
+    }
+
+    /// Append `segment` to the compressed data in the time series and return the size of `segment` in bytes.
+    pub fn append_segment(&mut self, segment: RecordBatch) -> usize {
+        let segment_size = Self::get_size_of_segment(&segment);
+
+        debug_assert!(
+            segment.schema() == Arc::new(Self::get_compressed_segment_schema()),
+            "Schema of record batch does not match compressed segment schema."
+        );
+
+        self.compressed_segments.push(segment);
+        self.size_in_bytes += segment_size;
+
+        segment_size
+    }
+
+    /// If the compressed segments are successfully saved to Parquet, return Ok, otherwise return Err.
```

Comment:
For consistency, should we include "Apache" when we reference an Apache project? At the moment it is inconsistent. Generally I think we should include it as part of the name.

Reply:
I concur.

Reply:
I agree. Should we only change it in comments or should we also change functions such as `save_to_parquet` to `save_to_apache_parquet`? Currently, I have just changed it in the comments.

Reply:
I suggest we change it for both comments and code to keep it consistent.

Reply:
Sounds good.

Reply:
I have changed it in the code. It should be noted that this means that `ArrowTimestamp` and `ArrowValue` should probably also be updated. There are also a few more places outside the storage engine where this needs to be changed, but I do not think this is the correct branch to do this in.

Reply:
I agree.